### DISCUSSION TITLE
Use bevy's rendering without implementing game logic in bevys ECS

### DISCUSSION THREAD

**Discussion by ju-manns** [bevyengine/bevy#11333] [Q&A]
Hi people! First time posting here :)

Is it possible to use bevys rendering functionality in a library-style way like the following?
``` rust
fn main() {
    let teapot = ... // create some teapot;

    let app = App::new()... // set up a bevy app with everything needed

    loop {
        teapot.transform = ... // change teapot location

        app.run(); // use app.run to draw the teapot. Maybe some parameters are needed here?
    }
}
```
As far as I understand the ECS so far, bevy generally takes ownership of its resources or requires references with static lifetimes.
Since I want to edit my teapot object inbetween draw calls inside the main function and not inside a bevy system, neither is an option.
Is there a good way to do this or am I going against bevy's premises here?

Thanks in advance for your help!

**Comment by bcolloran** [comment]
Hey @ju-manns, I don't know if this will be helpful or not, but you could try looking at this project:
https://github.com/fishfolk/bones

They have their own ecs but are using bevy for rendering. There may be some ideas in that project that could help you. Good luck!
